client/my-sites/controller.js

```diff
@@ -381,9 +382,22 @@ export function siteSelection( context, next ) {
 			return;
 		}
 	} else {
+		// Fetch the site from siteFragment.
+		dispatch( requestSite( siteFragment ) )
+			.then( () => {
+				const initialSiteId = getSiteId( getState(), siteFragment );
+
+				dispatch( setSelectedSiteId( initialSiteId ) );
+
+				if ( getSite( getState(), initialSiteId ) ) {
+					onSelectedSiteAvailable( context );
+				}
+			} )
+			.catch( noop );
```

Comment:
We don't want to do anything if we receive an error here. What we want is for Calypso to behave as standard: wait for all sites to be loaded before making decisions which site to select.